### ISSUE TITLE
Setting to make the inside of the keycap flat?

### ISSUE THREAD

**Issue by mvaneijgen** [riskable/keycap_playground#8]
This has to do with my previous topic https://github.com/riskable/keycap_playground/issues/4 but also solves an other problem I'm having. Is there a setting that makes the inside of the key flat? 

![Screen Shot 2022-09-17 at 10 11 34](https://user-images.githubusercontent.com/3217544/190847311-0d782eb8-6b93-44e8-a96f-f36374e42521.jpg)

I want to make some weird tilted shape key and I have this really cool wood filament, it has wood fibers in it, so I need to print it with a larger nozzle, the problem with that is that the stem is not that detailed and thus loses strength, so the thinking now is to print the stem separate with a smaller nozzle (and other material) this way I can print the keycap with as large a nozzle as I want, make the tilt as extreme as I want (because I don't have to worry about the stem poking through). Then just print a separate stem and glue it in place.

Here is where I got my inspiration from. 

![Screen Shot 2022-09-17 at 10 17 39](https://user-images.githubusercontent.com/3217544/190847484-3398ff1c-3bdb-4653-a0eb-4769aabe09e5.jpg)
[Screenshot from: Making an ALL WOOD Scrabble-themed keyboard — Evan and Katelyn ](https://www.youtube.com/watch?v=jSwMhR70wF8)

Any help is appreciated.

**Comment by riskable** [comment]
Yes!  This is what the `UNIFORM_WALL_THICKNESS` setting is for. If you set it to `false` that'll give you the flat-topped inside that you want 👍

Also, for this specific application I recommend trying out the snap-fit stem feature.  That way you can print the stem in something like PETG (which is much less fragile than wood filament) but keep the top in that nice wood filament.  It'll also let you change out the stem later if it breaks or becomes too loose over time (though I have yet to ever have to do that--even though I've been typing on such stems for almost a year now).

Aside: If you do go with the snap-fit stem I highly recommend using something like silicone caulk or B7000 adhesive to hold it in there (in addition to the snap-fit clips) because *it adds heft* and a bit of a shock-absorbing feature.  I don't know how to describe it but the snap-fit keycaps where I used silicone caulk or B7000 between the stem and the underside of the keycap just *feel* nicer than *just* snap-fit.  They also sound a bit better too (but that's much more subjective).  I'd imagine that it would soften the sound of hard wood-filled PLA quite a bit (so it wouldn't be as high pitched).

**Comment by AJolly** [comment]
@riskable btw, have you tried just printing them in TPU?  That's what I've done, and they are so much nicer on my poor finger joints.  Printed them with 1 wall line, zero infill.
